src/main/java/uk/gov/digital/ho/hocs/queue/migration/MigrationCaseData.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.digital.ho.hocs.queue.migration;
+
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+import com.jayway.jsonpath.ReadContext;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.digital.ho.hocs.client.casework.dto.ComplaintCorrespondent;
+import uk.gov.digital.ho.hocs.document.JSONToSimpleTextConverter;
+import uk.gov.digital.ho.hocs.queue.complaints.ComplaintData;
+import uk.gov.digital.ho.hocs.queue.complaints.CorrespondentType;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+public class MigrationCaseData implements ComplaintData {
+
+    public static final String CREATION_DATE = "$.creationDate";
+    public static final String COMPLAINT_TYPE = "$.complaint.complaintType";
+    public static final String APPLICANT_TYPE = "$.complaint.reporterDetails.applicantType";
+    public static final String APPLICANT_APPLICANT_NAME = "$.complaint.reporterDetails.applicantName";
+    public static final String APPLICANT_APPLICANT_EMAIL = "$.complaint.reporterDetails.applicantEmail";
+    public static final String APPLICANT_APPLICANT_PHONE = "$.complaint.reporterDetails.applicantPhone";
+    public static final String AGENT_APPLICANT_NAME = "$.complaint.reporterDetails.applicantDetails.applicantName";
+    public static final String AGENT_AGENT_NAME = "$.complaint.reporterDetails.agentDetails.agentName";
+    public static final String AGENT_AGENT_EMAIL = "$.complaint.reporterDetails.agentDetails.agentEmail";
+    public static final String AGENT_AGENT_PHONE = "$.complaint.reporterDetails.agentDetails.agentPhone";
+
+    private final ReadContext ctx;
+    private final String jsonBody;
+
+    public MigrationCaseData(String jsonBody) {
+        this.jsonBody = jsonBody;
+        ctx = JsonPath.parse(jsonBody);
+    }
+
+    @Override
+    public LocalDate getDateReceived() {
+        return LocalDate.parse(ctx.read(CREATION_DATE));
+    }
+
+    @Override
+    public String getComplaintType() {
```

Comment:
This is also the same as `UKVIComplaintData`.

Reply:
Fixed as suggested, extracted into an abstract class and inherited.